nixos/modules/programs/openvpn3.nix

```diff
@@ -1,43 +1,116 @@
 { config, lib, pkgs, ... }:
-
 let
+  json = pkgs.formats.json { };
   cfg = config.programs.openvpn3;
-in
-{
-  options.programs.openvpn3 = {
-    enable = lib.mkEnableOption "the openvpn3 client";
-    package = lib.mkOption {
-      type = lib.types.package;
-      default = pkgs.openvpn3.override {
-        enableSystemdResolved = config.services.resolved.enable;
+in {
+  options.programs.openvpn3 = let
+    inherit (lib)
+      mkEnableOption mkPackageOption mkOption literalExpression max options
+      lists;
+    inherit (lib.types) bool submodule ints;
+  in {
+    enable = mkEnableOption "the openvpn3 client";
+    package = mkPackageOption pkgs "openvpn3" { };
+    netcfg = mkOption {
+      description = "Network configuration";
+      default = { };
+      type = submodule {
+        options = {
+          settings = mkOption {
+            description =
+              "Options stored in {file}`/etc/openvpn3/netcfg.json` configuration file";
+            default = { };
+            type = submodule {
+              freeformType = json.type;
+              options = {
+                systemd_resolved = mkOption {
+                  type = bool;
+                  description = "Whether to use systemd-resolved integration";
+                  default = config.services.resolved.enable;
+                  defaultText =
+                    literalExpression "config.services.resolved.enable";
+                  example = false;
+                };
+              };
+            };
+          };
+        };
+      };
+    };
+    log-service = mkOption {
+      description = "Log service configuration";
+      default = { };
+      type = submodule {
+        options = {
+          settings = mkOption {
+            description =
+              "Options stored in {file}`/etc/openvpn3/log-service.json` configuration file";
```

Comment:
```suggestion
            description = "Options stored in {file}`/etc/openvpn3/log-service.json` configuration file";
```

Reply:
This is (and the following) formatting is by `nixfmt-rfc-style` which is used [specified][1] by `default.nix` so I am unsure if I should manually rollback the lines (gicen that it would move them back again). Correct me if I am wrong

[1]: https://github.com/NixOS/nixpkgs/blob/master/shell.nix#L28

Reply:
Let's follow the `nixfmt-rfc-style`